helm/kiam/templates/agent-daemonset.yaml

```diff
@@ -0,0 +1,132 @@
+{{- if .Values.agent.enabled -}}
+apiVersion: apps/v1beta2
+kind: DaemonSet
+metadata:
+  labels:
+    app: {{ template "kiam.name" . }}
+    chart: {{ template "kiam.chart" . }}
+    component: "{{ .Values.agent.name }}"
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: {{ template "kiam.fullname" . }}-agent
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "kiam.name" . }}
+      component: "{{ .Values.agent.name }}"
+      release: {{ .Release.Name }}
+      {{- if .Values.agent.podLabels }}
+{{ toYaml .Values.agent.podLabels | indent 6 }}
+      {{- end }}
+  template:
+    metadata:
+      {{- if .Values.agent.podAnnotations }}
+      annotations:
+{{ toYaml .Values.agent.podAnnotations | indent 8 }}
+      {{- end }}
+      labels:
+        app: {{ template "kiam.name" . }}
+        component: "{{ .Values.agent.name }}"
+        release: {{ .Release.Name }}
+      {{- if .Values.agent.podLabels }}
+{{ toYaml .Values.agent.podLabels | indent 8 }}
+      {{- end }}
+    spec:
+      hostNetwork: true
+      dnsPolicy: {{ .Values.agent.dnsPolicy }}
+      serviceAccountName: {{ template "kiam.serviceAccountName.agent" . }}
+    {{- if .Values.agent.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.agent.nodeSelector | indent 8 }}
+    {{- end }}
+      tolerations:
+{{ toYaml .Values.agent.tolerations | indent 8 }}
+      {{- if .Values.agent.affinity }}
+      affinity:
+{{ toYaml .Values.agent.affinity | indent 10 }}
+      {{- end }}
+      volumes:
+        - name: tls
+          secret:
+            {{- if .Values.agent.tlsSecret }}
+            secretName: {{ .Values.agent.tlsSecret }}
+            {{else}}
+            secretName: {{ template "kiam.fullname" . }}-agent
+            {{- end }}
+        - name: xtables
+          hostPath:
+            path: /run/xtables.lock
```

Comment:
Please add `type: FileOrCreate` to match https://github.com/uswitch/kiam/blob/master/deploy/agent.yaml#L32 and address https://github.com/helm/charts/issues/9261